File: Python/8kyu/8kyu-beginner-series-#4-cockroach.py
# Beginner Series #4 Cockroach
# Langugage: Python
# Version: 3.10
# Difficulty: 8 kyu

def cockroach_speed(s):
    return s // 0.036


# Tests
import codewars_test as test
from solution import cockroach_speed

@test.describe("Fixed Tests")
def fixed_tests():
    @test.it('Basic Test Cases')
    def basic_test_cases():
        test.assert_equals(cockroach_speed(1.08),30)
        test.assert_equals(cockroach_speed(1.09),30)
        test.assert_equals(cockroach_speed(0),0)

File: R/8kyu/8kyu-beginner-series-#4-cockroach.r
# Beginner Series #4 Cockroach
# Langugage: R
# Version: 3.4.1
# Difficulty: 8 kyu

# Solution 1
cockroach_speed <- function(s){
  floor(s * 27.7777777778)
}

# Solution (2 more concise)
cockroach_speed <- function(s){
  s %/% 0.036
}


# Tests
test_that('Basic tests', {
  expect_equal(cockroach_speed(1.08),30)
  expect_equal(cockroach_speed(1.09),30)
  expect_equal(cockroach_speed(0),0)
})